airflow/cli/commands/config_command.py

```diff
@@ -17,11 +17,54 @@
 """Config sub-commands"""
 import io
 
-from airflow.configuration import conf
+from configupdater import ConfigUpdater, NoSectionError
+
+from airflow.configuration import AIRFLOW_CONFIG, AirflowConfigException, conf
 
 
 def show_config(args):
     """Show current application configuration"""
     with io.StringIO() as output:
-        conf.write(output)
+        if args.section:
+            section = args.section
+            if conf.has_section(section):
+                conf._write_section(  # pylint: disable=protected-access
+                    fp=output,
+                    section_name=section,
+                    section_items=conf.getsection(section).items(),
+                    delimiter=" = "
+                )
+            else:
+                raise AirflowConfigException(f"No section '{section}'")
+        else:
+            conf.write(output)
         print(output.getvalue())
+
+
+def set_config_option(args):
+    """Set option in config"""
```

Comment:
Should we allow setting configs via cli ! ?

We already have 2 ways of setting configs.

- Environment Variables 
- Editing airflow.cfg

In my previous company too, we allowed users to access the airflow cli but we don't grant them perms to view/edit `airflow.cfg` file and we also hide access to configs in the UI via `expose_config` as it contains sensitive things like Fernet Key.

Reply:
If the user has access to a shell, he can also execute `cat ~/airflow/airflow.cfg` command or  `nano ~/airflow/airflow.cfg`.  I see no difference whether it will be available via friendly commands or the user will have to use a text editor.  If the user does not have permission to write to the file, this command will not work.

Reply:
These commands are just a friendlier alternative to editing the `airflow.cfg` file.

Reply:
> If the user has access to a shell, he can also execute `cat ~/airflow/airflow.cfg` command or `nano ~/airflow/airflow.cfg`. I see no difference whether it will be available via friendly commands or the user will have to use a text editor. If the user does not have permission to write to the file, this command will not work.

Not all, we set permission on the `airflow.cfg` file to not allow read and write.

Reply:
> These commands are just a friendlier alternative to editing the `airflow.cfg` file.

With the filesystem, we can set permissions the way we want. Allow/restrict access to certain files for users and groups. With the current CLI, it has no ACLs on it and would allow everyone to do everything.

Reply:
The main reason we had allowed access to CLI is to debug dags using list_dags command, similar to how Composer provides that using the following command:

```
gcloud composer environments run \
 test-environment --location us-central1 \
 list_dags -- -sd /home/airflow/gcs/data/test
```

Reply:
For example, check this popular Ansible role to install Airflow:

https://github.com/idealista/airflow-role/blob/master/tasks/config.yml#L23-L35

It uses **0640** permissions just for **airflow** user and **airflow** group and is run via systemd.

So no other user can edit/view this file.

Reply:
If the user who executes the command does not have permissions, the command will fail. This command edits the `airflow.cfg` file, so it requires identical permissions. You can change this requirement by changing the special privileges - chmod, but it is additionally configurable and rather no one does.
https://en.wikipedia.org/wiki/Setuid#SUID

Reply:
Ofcourse :man_facepalming:

Reply:
Ofcourse :man_facepalming: